tests/plugins/test_sendmail.py

```diff
@@ -122,6 +124,24 @@ def taskrelpath(self, task_id):
         return "tasks/%s" % task_id
 
 
+DEFAULT_ANNOTATIONS = {
+    'repositories': {
+        'unique': ['foo/bar:baz'],
+        'primary': ['foo/bar:spam'],
+    }
+}
+
+
+def mock_store_metadata_results(workflow, annotations=DEFAULT_ANNOTATIONS):
+    result = {}
+    result['annotations'] = {key: json.dumps(value) for key, value in annotations.items()}
+    workflow.exit_results[StoreMetadataInOSv3Plugin.key] = result
+
+
+def mock_store_missing_metadata_results(workflow):
+    workflow.exit_results[StoreMetadataInOSv3Plugin.key] = {}
```

Comment:
If we had one function to generate the annotations and one function to store them in the results, it might be easier to read in the actual test:
```
if metadata_stored:
    result = {}
    if annotations:
        result.update(make_annotations(workflow))
    mock_store_metadata_result(workflow, result)
```

and the regular case could set annotations by default:
```
mock_store_metadata_result(workflow)
```

Reply:
@twaugh what do you think of commit afb383e?

Reply:
Yes, I like it. :)